force-app/main/fslStatusTransition/classes/HOT_RequestStatusHandler.cls

```diff
@@ -120,41 +120,45 @@ public without sharing class HOT_RequestStatusHandler extends MyTriggers {
                 System.debug('numberOfPartiallyComplete: ' + numberOfPartiallyComplete);
                 System.debug('numberOfCannotComplete: ' + numberOfCannotComplete);
                 System.debug('numberOfCanceled: ' + numberOfCanceled);
-
-                if (numberOfScheduled > 0) {
-                    request.Status__c = statusMap.get('Scheduled');
-                }
-                if (numberOfDispatched == numberOfWorkOrder) {
-                    request.Status__c = statusMap.get('Dispatched');
-                    continue;
-                }
-                if (numberOfPartiallyComplete > 0) {
-                    request.Status__c = statusMap.get('Partially Complete');
-                }
-                if (
-                    numberOfCompleted > 0 &&
-                    numberOfCannotComplete > 0 &&
-                    (numberOfCannotComplete + numberOfCompleted == numberOfWorkOrder)
-                ) {
-                    request.Status__c = statusMap.get('Partially Complete');
-                    continue;
-                }
-                if (numberOfCompleted == numberOfWorkOrder) {
-                    request.Status__c = statusMap.get('Completed');
-                    continue;
-                }
-                if (numberOfCannotComplete == numberOfWorkOrder) {
-                    request.Status__c = statusMap.get('Cannot Complete');
-                    continue;
+                if (request.Status__c != 'Åpen') {
+                    if (numberOfNew > 0 || numberOfScheduled > 0) {
+                        request.Status__c = statusMap.get('New');
+                        continue;
+                    }
+                    if (
+                        numberOfDispatched > 0 &&
+                        (numberOfDispatched + numberOfCanceled + numberOfCannotComplete + numberOfCompleted +
+                        numberOfPartiallyComplete == numberOfWorkOrder)
+                    ) {
+                        request.Status__c = statusMap.get('Dispatched');
+                        continue;
+                    }
+                    if (numberOfInProgress > 0) {
```

Comment:
Denne vil bli "Pågår" om 1 er Pågår og 1 er "Dekket" f. eks. Er det ønsket? Statusflyt-diagrammet viser "Alle pågår" for "Pågår" status.